src/cluster/placement/algo/mirrored_test.go

```diff
@@ -697,6 +702,10 @@ func TestMirrorReplaceAndRevertBeforeCutover(t *testing.T) {
 	assert.NoError(t, placement.Validate(p))
 	assert.Equal(t, uint32(2), p.MaxShardSetID())
 
+	p, _, err = a.MarkAllShardsAvailable(p)
```

Comment:
Hmm doesn't this change the semantics of the test a bit? I haven't reasoned through it fully, but looking back at my draft PR I think I went through a good number of hoops to *not* change this part--what's the justification?